components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java

```diff
@@ -72,6 +75,27 @@ public SearchFacetImpl(final SearchService searchService,
   {
     this.searchService = checkNotNull(searchService);
     this.componentMetadataProducers = checkNotNull(componentMetadataProducers);
+    this.rebuildIndexLock = new ReentrantReadWriteLock();
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public void rebuildIndex() {
+    rebuildIndexLock.writeLock().lock();
```

Comment:
lock may be problematic in larger context of what we are going to be done in the near future, use with caution.